File: echo-guardian-production-foundation-v1.0/native/ios/EchoGuardianKit/Sources/EchoGuardianKit/Receipts/ProviderSandboxReceipt.swift
import Foundation

public struct ProviderSandboxReceipt: Codable, Equatable {
    public let schemaVersion: String
    public let provider: String
    public let providerMessageId: String
    public let deliveryStatus: String
    public let sandboxVerified: Bool
    public let sensitiveDetailsIncluded: Bool
    public let emergencyServicesUsed: Bool
}

public enum ProviderSandboxReceiptVerifier {
    public static func verify(provider: String, messageId: String, status: String, sandbox: Bool) -> ProviderSandboxReceipt? {
        guard sandbox == true else { return nil }
        guard status == "attempted" || status == "delivered" else { return nil }
        return ProviderSandboxReceipt(
            schemaVersion: "0.7",
            provider: provider,
            providerMessageId: messageId,
            deliveryStatus: status,
            sandboxVerified: true,
            sensitiveDetailsIncluded: false,
            emergencyServicesUsed: false
        )
    }
}

File: echo-guardian-production-foundation-v1.0/native/android/echo-guardian-core/src/main/java/org/echoguardian/receipts/ProviderSandboxReceipt.kt
package org.echoguardian.receipts

data class ProviderSandboxReceipt(
    val schemaVersion: String = "0.7",
    val provider: String,
    val providerMessageId: String,
    val deliveryStatus: String,
    val sandboxVerified: Boolean = true,
    val sensitiveDetailsIncluded: Boolean = false,
    val emergencyServicesUsed: Boolean = false
)

object ProviderSandboxReceiptVerifier {
    fun verify(provider: String, messageId: String, status: String, sandbox: Boolean): ProviderSandboxReceipt? {
        if (!sandbox) return null
        if (status != "attempted" && status != "delivered") return null
        return ProviderSandboxReceipt(provider = provider, providerMessageId = messageId, deliveryStatus = status)
    }
}
